components/plugin_marketplace/marketplace_usage_message.tsx

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import {FormattedMessage, useIntl} from 'react-intl';
+
+import {PreferenceType} from '@mattermost/types/preferences';
+
+import {getCloudLimits} from 'mattermost-redux/selectors/entities/cloud';
+import {getIntegrationsUsage} from 'mattermost-redux/selectors/entities/usage';
+import {cloudFreeEnabled, getMyPreferences} from 'mattermost-redux/selectors/entities/preferences';
+import {getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
+import {getPreferenceKey} from 'mattermost-redux/utils/preference_utils';
+
+import {savePreferences} from 'mattermost-redux/actions/preferences';
+
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import CloseIcon from 'components/widgets/icons/close_icon';
+import CloudStartTrialButton from 'components/cloud_start_trial/cloud_start_trial_btn';
+
+const HIDE_USAGE_MESSAGE_PREFERENCE_CATEGORY = 'usage_limits';
+const HIDE_USAGE_MESSAGE_PREFERENCE_NAME = 'hide_marketplace_usage_message';
+
+const usePreferenceCheck = (): [boolean, (value: boolean) => void] => {
```

Comment:
I love this hook ❤️ ! but wondering if maybe it should go in its own file? there is a common/hooks folder, where this one could live. Also, I think this one with some little enhancements could be a general hook one to retrieve/save a preference.